### ISSUE TITLE
explode() expects parameter 2 to be string

### ISSUE THREAD

**Issue by umairali** [anlutro/php-curl#25]
I am using laravel 4.2 and i am facing this error in response 

 $file = curl_file_create($sqlFileName);
 $response = cURL::rawPost($url, ['sqlfile' => $file]);
 return $response;

{"error":{"type":"ErrorException","message":"explode() expects parameter 2 to be string, array given","file":"D:\\htdocs\\sahilwoodtest\\vendor\\anlutro\\curl\\src\\Response.php","line":123}}

**Comment by anlutro** [comment]
That's odd, it looks like the server you're sending the request to is sending duplicate HTTP status headers. Could you run the following code and show me the output? Remove any sensitive data if necessary.

```php
$file = curl_file_create($sqlFileName);
$response = cURL::newRawRequest('post', $url, ['sqlfile' => $file])
    ->setOption(CURLOPT_VERBOSE, true)
    ->send();
```

**Comment by umairali** [comment]
I am sending request from ajax 

![capture](https://cloud.githubusercontent.com/assets/5318005/8905659/8420f410-3481-11e5-96e6-967d8c9e3e92.PNG)

![capture1](https://cloud.githubusercontent.com/assets/5318005/8905666/8fd04bf8-3481-11e5-8e30-12613adbcabd.PNG)

**Comment by umairali** [comment]
After using this

$file = curl_file_create($sqlFileName);
$response = cURL::newRawRequest('post', $url, ['sqlfile' => $file])
    ->setOption(CURLOPT_VERBOSE, true)
    ->send();


{"error":{"type":"InvalidArgumentException","message":"Method [http:\/\/testsaahil.saahil.digitalmanager.pk\/syncdb\/localtoserverdbuploadandsync] not a valid HTTP method.","file":"D:\\Dropbox\\Digital Softs\\htdocs\\sahilwoodtest\\vendor\\anlutro\\curl\\src\\Request.php","line":105}}

**Comment by anlutro** [comment]
Which version of the package are you on? If you're on 1.x you could try updating to 1.3, there shouldn't be any breaks.

**Comment by umairali** [comment]
i m using 1.3 version

**Comment by umairali** [comment]
"anlutro/curl": "1.3.0"

**Comment by anlutro** [comment]
Ah, my bad, I misread something.

Are you sure you're doing `newRawRequest('post', $url, ['sqlfile' => $file])`? From the error it looks like you may have switched the `'post'` and `$url` arguments.

If you could find the error stack trace that would be great as well. It should be in `app/storage/logs/apache2handler.log` by default.

**Comment by skvn** [comment]
The reason is wrong way to parse headers (Response.php:105)
Please, try response with headers
X-Var: A
X-Var: B

And then response with headers
X-Var: A
X-Var: B
X-Var: C

You will have different and strange behaviour for second case.

As of described issue the problem is curl sent header 'Expect: 100-continue' for POST request with over 1024 bytes size.
The response will be HTTP/1.1 100 Continue\r\n\r\nHTTP/1,1 200OK\r\n....

Workaround: call setHeader('Expect', '') for Request instance.

**Comment by anlutro** [comment]
Aha, that answers my question. I wasn't aware of the Expect: 100-continue stuff - I didn't know if there was an actual legit reason to have duplicate HTTP status headers, which is why I hesitated to make any fixes.

You're also right that more than 2 headers of the same key produce the wrong results - this is unrelated to the original issue but I will definitely have that fixed as well.

**Comment by anlutro** [comment]
This should be fixed in 1.3.2. Thanks for the help!
